.github/workflows/ci.yml

```
@@ -0,0 +1,55 @@
+
+      - name: Run tests and generate JaCoCo report
+        run: ./gradlew clean test jacocoTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          files: ./build/reports/jacoco/test/jacocoTestReport.xml
+          verbose: true
+
+      - name: Report test Coverage to PR
+        id : jacoco
```

YAMLlint (1.37.1)

[warning] too many spaces before colon

(colons)